hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/RegionSizeCalculator.java

```diff
@@ -96,11 +98,14 @@ private void init(RegionLocator regionLocator, Admin admin) throws IOException {
   }
 
   private Set<ServerName> getRegionServersOfTable(RegionLocator regionLocator) throws IOException {
-
     Set<ServerName> tableServers = Sets.newHashSet();
-    for (HRegionLocation regionLocation : regionLocator.getAllRegionLocations()) {
-      tableServers.add(regionLocation.getServerName());
-    }
+
+    // The region locations could contain `null` ServerName instances if the region is currently
+    // in transition, we filter those out for now, which impacts the size calculation for these
+    // regions temporarily until the ServerName gets filled in later
+    tableServers.addAll(regionLocator.getAllRegionLocations().stream()
```

Comment:
rather than collect to a list just to add to a set, you could collect directly to a set

Reply:
Fixed, thanks.